src/array.cr

```diff
@@ -1808,6 +1811,21 @@ class Array(T)
     insert 0, obj
   end
 
+  # Prepend multiple values. The same as `unshift`, but takes an arbitrary number
+  # of values to add to the array. Returns `self`.
+  def unshift(*values : T)
+    new_size = @size + values.size
+    resize_to_capacity(Math.pw2ceil(new_size)) if @size > @capacity
+    move_value = values.size
+    @buffer.move_from(@buffer - move_value, @size + move_value)
```

Comment:
`move_to` and `move_from` are "the same", it's just where you want to focus semantically. But what was wrong were the positions of the move.